src/Bounds.cpp

```diff
@@ -111,6 +111,15 @@ class Bounds : public IRVisitor {
     Bounds(const Scope<Interval> *s, const FuncValueBounds &fb, bool const_bound)
         : func_bounds(fb), const_bound(const_bound) {
         scope.set_containing_scope(s);
+
+        // Find any points that are single_points but fail is_single_point due to
+        // pointer equality checks and replace with single_points.
+        for (auto item = s->cbegin(); item != s->cend(); ++item) {
+            const Interval &item_interval = item.value();
+            if (!item_interval.is_single_point() && item_interval.is_bounded() && can_prove(item_interval.min == item_interval.max)) {
```

Comment:
nit: we have no line length limit but this one is long enough to deserve a line break, IMHO

Reply:
fixed in 7a0cc0c